### PR TITLE
Increase coverage of the Core folder

### PR DESCRIPTION
**Pull request by byorda-glb**
### Purpose

Increase code coverage on the Core folder

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

Aaron Tang
Michael Kirschner

### FYIs

Alfredo Pozo

### CODE REVIEW

**Comment by QilongTang** [comment]
LGTM, Can you also post the coverage difference for future reference?

**Comment by byorda-glb** [comment]
> LGTM, Can you also post the coverage difference for future reference?

This is what I am seeing in my report:

image

But this is not accurate. DotCover groups by namespace, and the classes I'm covering here, even though their files are located in the Core folder, do not belong to the Dynamo.Core namespace.

File: test/DynamoCoreTests/Core/AssemblyLoadedEventArgsTests.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    public class AssemblyLoadedEventArgsTests
+    {
+        [Test]
+        [Category("UnitTests")]
+        public void AssemblyLoadedEventArgsConstructorTests()
+        {
+            var assemblyMock = Assembly.GetExecutingAssembly();
+            AssemblyLoadedEventArgs eventArgs = new AssemblyLoadedEventArgs(assemblyMock);
+
+            Assert.AreEqual(assemblyMock, eventArgs.Assembly);
```
Comment (QilongTang) [inline test/DynamoCoreTests/Core/AssemblyLoadedEventArgsTests.cs:17]: How about the code coverage under NodeModelAssemblyLoader? Is there a task to cover that or will be included in follow up PRs?

Comment (byorda-glb) [reply test/DynamoCoreTests/Core/AssemblyLoadedEventArgsTests.cs:17]: NodeModelAssemblyLoader contains the class AssemblyLoadedEventArgs. So this is a small increment in the coverage of the former.

Comment (QilongTang) [reply test/DynamoCoreTests/Core/AssemblyLoadedEventArgsTests.cs:17]: @byorda-glb Understood. I mean do you plan to add coverage to `NodeModelAssemblyLoader` later?

Comment (byorda-glb) [reply test/DynamoCoreTests/Core/AssemblyLoadedEventArgsTests.cs:17]: I don't think so. The code that remains uncovered is composed of events and catch blocks, which are very tricky to test. If we need more coverage in Core, I'd look elsewhere first.

Comment (QilongTang) [reply test/DynamoCoreTests/Core/AssemblyLoadedEventArgsTests.cs:17]: @byorda-glb Thank you for clarification. I think this PR is relatively safe to go in. I will merge it as it but @mjkkirschner may contact on fixing some previous tests

**Comment by QilongTang** [comment]
LGTM

**Pull request merged**